### PR TITLE
`rz_utf8_decode()`: Declare overlong encodings as invalid UTF-8

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr makes `rz_utf8_decode()` declare [overlong encodings](https://en.wikipedia.org/wiki/UTF-8#Overlong_encodings) as invalid UTF-8, since they break the 1-to-1 relationship between byte sequences and Unicode code points.

**Test plan**

All builds are green.

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

File: test/db/tools/rz_bin [hunk 1]
```diff
@@ -176,15 +176,15 @@ NAME=rz-bin -zz pe
 FILE=bins/pe/ioli/w32/crackme0x00.exe
 CMDS=!rz-bin -zz ${RZ_FILE} | grep "Password:"
 EXPECT=<<EOF
-86  0x00002619 0x00404019 10  11   .rdata  ascii   Password: 
+81  0x00002619 0x00404019 10  11   .rdata  ascii   Password:
```
Comment (ret2libc) [inline test/db/tools/rz_bin:179]: why is this changed? i guess it's because some other strings are removed from the output? Can you check what's happening to those other strings and if they are correctly removed?

Comment (kazarmy) [reply test/db/tools/rz_bin:179]: ok

```bash
rz-bin -zz bins/pe/ioli/w32/crackme0x00.exe | cut -c 5- > before/after.txt
diff -u before.txt after.txt
```
```patch
--- before.txt  2021-09-29 00:01:31.143047000 +0800
+++ after.txt   2021-09-29 00:02:12.623047000 +0800
@@ -9,7 +9,6 @@
 0x00000218 0x00000218 6   7            ascii   .idata
 0x0000040a 0x0040100a 5   7    .text   utf8    E\b1ۉu blocks=Basic Latin,Arabic
 0x00000431 0x00401031 4   6    .text   utf8    1҉T$ blocks=Basic Latin,Cyrillic
-0x000004eb 0x004010eb 4   6    .text   utf8    1\tD$
 0x000006a2 0x004012a2 4   5    .text   ascii   \r a@
 0x000006cd 0x004012cd 4   5    .text   ascii   $`A@
 0x000006f8 0x004012f8 4   5    .text   ascii   $`A@
@@ -20,17 +19,14 @@
 0x00000921 0x00401521 6   8    .text   utf8    Z[^_]É blocks=Basic Latin,Latin-1 Supplement
 0x0000099a 0x0040159a 5   6    .text   ascii   \b[^_]
 0x000009aa 0x004015aa 4   5    .text   ascii   M\b<P
-0x00000a88 0x00401688 5   7    .text   utf8    1\tD$\f
 0x00000af8 0x004016f8 4   6    .text   utf8    ]ÍC  blocks=Basic Latin,Latin-1 Supplement
-0x00000c49 0x00401849 8   10   .text   utf8    D$\f1\tD$\b
+0x00000c49 0x00401849 4   5    .text   ascii   D$\f1
 0x00000d16 0x00401916 4   6    .text   utf8    [^]Í blocks=Basic Latin,Latin-1 Supplement
-0x00000d5f 0x0040195f 5   7    .text   utf8    1\tD$\b
 0x00000d7f 0x0040197f 4   6    .text   utf8    [^]ø blocks=Basic Latin,Latin-1 Supplement
 0x00000db8 0x004019b8 4   6    .text   utf8    1ҋE\f blocks=Basic Latin,Cyrillic
 0x00000e2b 0x00401a2b 4   5    .text   ascii   [^_]
 0x00000e91 0x00401a91 4   5    .text   ascii   9Z\fu
 0x00000eae 0x00401aae 7   9    .text   utf8    [^_]É<$ blocks=Basic Latin,Latin-1 Supplement
-0x00000f42 0x00401b42 5   7    .text   utf8    1\tD$\b
 0x00001050 0x00401c50 5   6    .text   ascii   ~\tzt\a
 0x00001057 0x00401c57 5   7    .text   utf8     [^]É blocks=Basic Latin,Latin-1 Supplement
 0x00001093 0x00401c93 14  15   .text   ascii   <Rt+<Pt\b<Lu(AC
@@ -49,7 +45,7 @@
 0x0000174a 0x0040234a 4   6    .text   utf8    MԋY\f blocks=Basic Latin,Cyrillic Supplement
 0x00001752 0x00402352 4   6    .text   utf8    uЉ4$ blocks=Basic Latin,Cyrillic
 0x000017a7 0x004023a7 5   8    .text   utf8    MԋE̋Q blocks=Basic Latin,Cyrillic Supplement,Combining Diacritical Marks
-0x00001899 0x00402499 8   12   .text   utf8    ωM\rp$9ӉU blocks=Greek and Coptic,Basic Latin,Cyrillic
+0x0000189e 0x0040249e 5   7    .text   utf8    p$9ӉU blocks=Basic Latin,Cyrillic
 0x000018e5 0x004024e5 4   6    .text   utf8    Mԉ\f$ blocks=Basic Latin,Cyrillic Supplement
 0x00001933 0x00402533 5   7    .text   utf8    J\f1ҋy blocks=Basic Latin,Cyrillic
 0x00001956 0x00402556 4   5    .text   ascii   s\f9E
@@ -59,7 +55,6 @@
 0x00001a36 0x00402636 6   9    .text   utf8    EԋMЋP\f blocks=Basic Latin,Cyrillic Supplement,Cyrillic
 0x00001a48 0x00402648 9   12   .text   utf8    |[^_]ËuԉV blocks=Basic Latin,Latin-1 Supplement,Cyrillic Supplement
 0x00001aaa 0x004026aa 4   5    .text   ascii   \bB;]
-0x00001aaf 0x004026af 5   8    .text   utf8    rߋE\vu blocks=Basic Latin,NKo
 0x00001b7a 0x0040277a 4   5    .text   ascii   \bA;]
 0x00001cc1 0x004028c1 7   9    .text   utf8    [^_]É4$ blocks=Basic Latin,Latin-1 Supplement
 0x00001dc6 0x004029c6 4   6    .text   utf8    1ۉT$ blocks=Basic Latin,Arabic
```

all of them appear to be false positives so i suppose there's nothing to be worried about

**Review by ret2libc — APPROVED** [review]
